tests/model_forms/test_modelchoicefield.py

```diff
@@ -55,6 +55,20 @@ def test_basics(self):
         with self.assertRaisesMessage(ValidationError, msg):
             f.clean(c4.id)
 
+        # Clean with model instance.
+        self.assertEqual(f.clean(self.c1), self.c1)
+
+        # Clean with instance of different model.
+        book = Book.objects.create()
+        msg = "['Select a valid choice. That choice is not one of the available choices.']"
+        with self.assertRaisesMessage(ValidationError, msg):
+            f.clean(book)
+
+        # Clean when using to_field_name.
+        f = forms.ModelChoiceField(Category.objects.all(), to_field_name="slug")
+        self.assertEqual(f.clean(self.c1.slug), self.c1)
```

Comment:
This one is strictly extra, since it's adding a test for the _normal™_ `to_field_name` case, which didn't seem to have a test.

Reply:
Should we commit that in another PR ?

Reply:
Perhaps just a second commit. 

`Added additional test for ModelChoiceField.to_field_name usage.`

Reply:
Hi Carlton,

I did split the commit and pushed it.

Etienne